### ISSUE TITLE
Time Zone Discrepancy

### ISSUE THREAD

**Issue by tomtjes** [TandoorRecipes/recipes#2622]
### Tandoor Version

latest

### Setup

Docker / Docker-Compose

### Reverse Proxy

SWAG

### Bug description

I noticed, when I create a cook log entry the time displays correctly, e.g. as
<img width="520" alt="Screenshot 2023-09-06 at 20 31 42" src="https://github.com/richardtallent/vue-simple-calendar/assets/7606307/d5e01c54-898b-4f20-9b8b-a04fa28a11d9">
Note, I'm not in the same timezone as the server. This shows when I look at the actual timestamp of the cook log:
<img width="349" alt="Screenshot 2023-09-06 at 20 32 31" src="https://github.com/richardtallent/vue-simple-calendar/assets/7606307/a115633c-350f-4a15-b06c-df5b17874158">
It's in "+02:00", while I'm in "-04:00".
I'm trying to automate some things with cook logs using the API, which is how I ran into this issue.

**Comment by smilerz** [comment]
Makes sense - the server you are posting to is +2.  

The front end doesn't provide any timezone information.  This is probably the only place that times are exposed to end users for editing.

**Comment by tomtjes** [comment]
Correct, in all other places I can think of it's the date only, without the time. So no issue there. But if I can enter a time here, I guess it should be saved correctly. I don't know much about the front end, but it should be possible to figure out the difference between local time and server time to create an accurate time stamp, right?

This could however affect search, for instance, where only the date is taken into account. In the example above, the expected cook log timestamp would be `2023-09-07T02:31:00+02:00`. If I'm in -04:00 and search for anything cooked on or before 2023-09-06 this wouldn't show up, because in server time it was cooked on the 7th. So search would need to internally work with time as well and convert `2023-09-06` in -04:00 local time to `2023-09-07T06:00:00+02:00` for the "before date" search and `2023-09-05T18:00:00+02:00` for the "after date" search.

**Comment by vabene1111** [comment]
interesting, working with timezones is always a bit difficult, but as you said, if there are times they should be correct. 

The question for me is: are times actually needed here or should we maybe convert this to a date only field? 

If not we need to make the frontend timezone aware and send that information to the server to convert it to the correct time in the background, that should probably work as well as the browser should be able to send a timezones timestamp

**Comment by smilerz** [comment]
With a 6-hour difference between client and server in this example, even just dates will have discrepancies.

Cooklog and MealPlan would be the only areas impacted by this I think?

**Comment by vabene1111** [comment]
shopping shows the timestamps when something was added but thats handled automatically by the database so that should be fine. Invite Links might be time zone sensitive as well although valid until is likely always set with some spare time

**Comment by tomtjes** [comment]
> are times actually needed here or should we maybe convert this to a date only field?

I actually find the time distracting. I don't need the information that I cooked something at 20:31 (and I'm too lazy to correct the pre-filled field to the actual time I cooked the dish). Maybe other people have use for that information.

(I have started to work on a tool that imports cook logs into a calendar app, which creates events at the given timestamp, but still...)

> With a 6-hour difference between client and server in this example, even just dates will have discrepancies.

If you'd just pick the date, say `2023-09-06` and save that to the database, it doesn't matter what time zone the server or the user are in. The user cooked it on that day and that is saved to the cook log.

**Comment by smilerz** [comment]
> (I have started to work on a tool that imports cook logs into

The database stores as a datetime (as all of the created_at fields do).  We could change that specific field to just a date field - but it would be inconsistent with every other model.  One option would be to move the log date to a separate field.

**Comment by vabene1111** [comment]
yes i am leaning towards that. have created_at the way it is in all models and an additonal `cook_date` that contains what was previously the created_at value. 

Then we just use a simple data migration for all old entries.

**Comment by anytimesoon** [comment]
The timezone discrepancy is now apparent when making the meal plan request on the homepage dashboard.

My server is in a timezone several hours later than where I am. This results in "today's" recipe not showing in the dashboard. I need to go to the mealplan view.

When the mealplan view loads, there appears to be two calls. One for this week's recipes (which loads without "todays"), and another for everthing else (here is where "today's") finally loads.

**Comment by vabene1111** [comment]
thanks for the report, I thought i fixed this with V2 but maybe I missed something

**Comment by anytimesoon** [comment]
Stand down. It started working today after the update to 2.3

I'll keep an eye on it. It's been a bit intermittent. Maybe it's not timezone related? I just assumed that would be the issue.

**Comment by vabene1111** [comment]
there were some issues with the call I did fix, not sure if it was in 2.3 but ok, closing for now, please re open if something comes up again.
